### PR TITLE
Feature/server tests

### PR DESCRIPTION
**Pull request by FelipeParreira**
I implemented the tests for the API using JEST. I still couldn't fully integrate it with Circle CI, because I still need to find out how to make a server and a DB run with Circle CI, but all my tests do pass locally in my machine.

### CODE REVIEW

**Comment by supersilent** [comment]
My suggestions are here.
1. Removing commented out codes 
2. Removing log files.
npm-debug.log.*
3. Removing unrelated files/codes.
Specs/Server/server.test.js
Specs/sum.js
Specs/sum.test.js
Specs/t.js
cci-config-generator.sh ?
4. Removing unneeded console.log

File: .eslintrc.js [hunk 1]
```diff
@@ -6,5 +6,15 @@
...
-  "extends": "airbnb-base"
+  "parser": "babel-eslint",
+  "extends": "airbnb-base",
+  "env": {
+    "browser": true,
+    "node": true
+  },
+  "plugins": [
+    "react",
+    // "jsx-ally",
```
Comment (supersilent) [inline .eslintrc.js]: Is this needed?

Comment (supersilent) [inline .eslintrc.js:17]: Is this needed?

File: Specs/Server/server.test.js [hunk 2]
```diff
@@ -0,0 +1,266 @@
+/* eslint no-undef: 0 */ // --> OFF
+const request = require('request-promise');
+const { toBeType } = require('jest-tobetype');
+const fetch = require('node-fetch');
+
+expect.extend({ toBeType });
+
+describe('test', () => {
```
Comment (supersilent) [inline Specs/Server/server.test.js:8]: It would be better to check the result specifically, not only the status code.

File: Specs/Server/server.test.js [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+  test(async () => {
+    await request.post('http://localhost:3000/test')
+      .on('response', (response) => {
+        expect(response.statusCode).toBe(201);
+      });
+  });
+});
+
+// describe('GET all questions', async () => {
```
Comment (supersilent) [inline Specs/Server/server.test.js]: It would be better to remove commented out code.

> Please, don’t commit commented out code
> https://blog.kentcdodds.com/please-don-t-commit-commented-out-code-53d0b5b26d5f

File: Specs/sum.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+function sum(a, b) {
```
Comment (supersilent) [inline Specs/sum.js]: Is this function needed for the project?

File: Specs/sum.test.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+/* eslint no-undef: 0 */ // --> OFF
+const sum = require('./sum');
+
+test('adds 1 + 2 to equal 3', () => {
```
Comment (supersilent) [inline Specs/sum.test.js]: Is this test needed for the project?

File: Specs/t.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
+// /* eslint import/no-extraneous-dependencies: 0 */ // --> OFF
```
Comment (supersilent) [inline Specs/t.js]: Is this file needed for the project?

File: cci-config-generator.sh [hunk 7]
```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+
+# Script to generate a CircleCI 2.0 `.circleci/config.yml` file
+# Please see the README for more details
```
Comment (supersilent) [inline cci-config-generator.sh]: If this is not needed to run the app, this should be added to .gitignore maybe.

File: client/src/Components/App.jsx [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    this.state = {
+      questions: [],
+    };
+  }
+
+  render() {
+    return (
+      <div>
+        <input type="text" class="form-control"/>
+        <QuestionList questions={this.state.questions} hotelId={4}/>
```
Comment (supersilent) [inline client/src/Components/App.jsx:16]: >`hotelId={4}`

Is hotelId always 4?

File: client/src/Components/QuestionList.jsx [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+      questions: [],
+    };
+  }
+
+  componentDidMount() {
+    $.ajax({
+      type: 'GET',
+      url: `http://localhost:3000/hotels/${this.props.hotelId}/questions`,
+      success: (questions) => {
+        console.log('questions', questions);
```
Comment (supersilent) [inline client/src/Components/QuestionList.jsx:18]: > `console.log('questions', questions);`

Is this needed in the production environment?

File: npm-debug.log.38613720 [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+7 error Darwin 17.6.0
+8 error argv "/Users/lucasparreiradefariaborges/.nvm/versions/node/v6.13.1/bin/node" "/Users/lucasparreiradefariaborges/.nvm/versions/node/v6.13.1/bin/npm" "ls" "--depth" "0" "--json"
+9 error node v6.13.1
+10 error npm  v3.10.10
+11 error code EPIPE
+12 error errno EPIPE
+13 error syscall write
+14 error write EPIPE
+15 error If you need help, you may report this error at:
+15 error     <https://github.com/npm/npm/issues>
```
Comment (supersilent) [inline npm-debug.log.38613720]: I thinks these files should be .gitignored and not be commited.

File: package.json [hunk 11]
```diff
@@ -8,31 +8,51 @@
...
   },
   "devDependencies": {
+    "@babel/core": "^7.2.2",
+    "babel-core": "^7.0.0-bridge.0",
+    "babel-eslint": "^10.0.1",
+    "babel-jest": "^23.6.0",
     "eslint": "^5.10.0",
+    "eslint-config-airbnb": "^17.1.0",
     "eslint-config-airbnb-base": "^13.1.0",
     "eslint-config-hackreactor": "git://github.com/reactorcore/eslint-config-hackreactor",
```
Comment (supersilent) [inline package.json:43]: `eslint-config-airbnb` and `eslint-config-airbnb-base` and `eslint-config-hackreactor` are used for the project?

File: server/query.js [hunk 12]
```diff
@@ -71,7 +71,7 @@ const getAllQuestions = (HotelID, res) => {
         questionData[i].AnswersUsers = users.splice(0, questionData[i].Answers.length);
       }
       console.log('question', questionData[0]);
```
Comment (supersilent) [inline server/query.js]: `console.log` should be removed if that is for a test.

**Comment by FelipeParreira** [comment]
@supersilent, I made the necessary changes. Only one thing, the server.test.js contains my tests for the my server (they are running locally, but I still haven't found out a way to make them run on circle ci).
Hope you can approve my PR now ;)

**Review by supersilent — APPROVED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (FelipeParreira) [reply .eslintrc.js:17]: I found out that we do; otherwise, the linter would throw us errors on anything we wrote using JSX.

File: Specs/Server/server.test.js (hunk 2 above)
Comment (FelipeParreira) [reply Specs/Server/server.test.js:8]: I understand your concern, but in this case this functionality is not implemented fully in my server (it is outside of the scope of my service); therefore, I just have a stub route to deal with it; someone can implement this feature in the future if they want.

File: Specs/Server/server.test.js (hunk 3 above)
Comment (FelipeParreira) [reply Specs/Server/server.test.js]: I uncommented those lines already :)

File: Specs/sum.js (hunk 4 above)
Comment (FelipeParreira) [reply Specs/sum.js]: Just a sample test; it is already in the .gitignore file now.

File: client/src/Components/App.jsx (hunk 8 above)
Comment (FelipeParreira) [reply client/src/Components/App.jsx:16]: No, I needed to have a specific hotel number in order to make my API calls. When we integrate our components we need to have a way to provide me (and all of us) with the number of the hotel the user is checking out in the moment.

File: package.json (hunk 11 above)
Comment (FelipeParreira) [reply package.json:43]: The instructions on Learn seem to tell us to use only the AirBnB style guide, but I am not sure if we can get rid of the HR style guide entirely.

File: client/src/Components/QuestionList.jsx (hunk 9 above)
Comment (FelipeParreira) [reply client/src/Components/QuestionList.jsx:18]: I have removed the console.log in my newest branch ;)
